### PR TITLE
[HttpFoundation] Use relative timestamps with MemcachedSessionHandler

### LINKED ISSUE #48619: Memcached session handler issue with clock skew
### Symfony version(s) affected

All versions

### Description

In https://github.com/symfony/symfony/commit/0216e056051a6304cab01defc107fb583f3f5efa the TTL values switch from relative time to absolute timestamps. I wonder why this change was made.

"Memcached implements a monotonic clock to ensure it always goes forward generally when time goes forward for the computer."
https://github.com/memcached/memcached/issues/278

But the internal clock can have skew in some environments. I encountered a Debian system running on VMware ESX host that actually has this. So while the OS clock is correct, the internal memcached clock is not (can be checked with the memcached stats command).

If the time drifts too far, memcached will see that entries are expired at the point you are adding them, and not add them to the store. So, your sessions are not created, logins will fail, carts can not be created, ...

As suggested here https://github.com/memcached/memcached/issues/486 it "relative expiration will still work as advertised since the drift is slow".

So while this seems like an edge case and maybe should be checked on the the infrastructure level instead of in this code, it does feel like this would fix it in all situations.

### How to reproduce

Only reproducible on a system that has clock skew on the memcached monotonic timer.

```
$ date
Mon 12 Dec 2022 08:18:37 PM CET

$ echo stats | nc 127.0.0.1 11211 | grep time
STAT time 1670876432

$ date -d @1670876432
Mon 12 Dec 2022 09:20:32 PM CET
```

### Possible Solution

If the time() calls are removed from the MemcachedSessionHandler that would result in relative TTL's and clock skew will no longer be an issue. I'll do a MR if we agree to fix this.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by tvlooy**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | 
| Deprecations? |
| Tickets       | Fix #48619
| License       | MIT
| Doc PR        | 

Use relative timestamps. If for some reason your client is not in sync with the server clock, you at least agree about the relative TTL. Or see #48619 for a case where the memcached monotonic clock is going faster than the server and causes users to add entries that are expired on entry.

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MemcachedSessionHandler.php [hunk 1]
```diff
@@ -77,7 +77,7 @@ protected function doRead(string $sessionId)
     #[\ReturnTypeWillChange]
     public function updateTimestamp($sessionId, $data)
     {
-        $this->memcached->touch($this->prefix.$sessionId, time() + (int) ($this->ttl ?? \ini_get('session.gc_maxlifetime')));
```
Comment (derrabus) [inline src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MemcachedSessionHandler.php:80]: I've read up how PHP's memcached extension interprets the values passed here: https://www.php.net/manual/en/memcached.expiration.php

tl;dr: If the passed integer is < 30 days in seconds, the value is considered to be a relative TTL. Otherwise, the value is assumed to be a UNIX timestamp.

My guess is that we added `time()` to be sure our relative TTL is never mistaken as a timestamp.

If we go back to relative TTLs again, I suggest we make sure our TTL is below that threshold and continue to add `time()` otherwise.

Comment (tvlooy) [reply src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MemcachedSessionHandler.php:80]: yes indeed https://github.com/memcached/memcached/blob/master/doc/protocol.txt#L139

**Comment by nicolas-grekas** [comment]
Thank you @tvlooy.
